HelpDeskCommonMethods/src/main/java/webdriverbase/AppPage.java

```
@@ -0,0 +1,145 @@
+	public static String WINDOWS_PATH_TO_TEST_DATA_DIR = "src/main/resources/";
+	public static int WAIT_TIME_SEC = 60;	
+	protected WebDriver driver ;
+	
+	JavascriptExecutor javaScriptExecutor;
+	
+	public AppPage(WebDriver driver) {
+		this.driver = driver;
+		waitImplicitly();	
+		PageFactory.initElements(driver, this);
+		maximizeWindow();
+	}
+	
+	public WebDriver getDriver() {
+		return this.driver;
+	}
+
+	public void get(String url) {
+		this.driver.get(url);
+	}
+
+	public String getCurrentUrl() {
+		return this.driver.getCurrentUrl();
+	}
+	
+	public void maximizeWindow() {
+			driver.manage().window().maximize();		
+	}
+	
+	public void waitImplicitly() {
+		driver.manage().timeouts().implicitlyWait(WAIT_TIME_SEC, TimeUnit.SECONDS);
+	}
+
+	public void waitImplicitly(int timeOutInSeconds) {
+		driver.manage().timeouts().implicitlyWait(timeOutInSeconds, TimeUnit.SECONDS);
+	}
+	
+	public void clearAndType(WebElement element, String text) {
+		element.clear();
+		element.sendKeys(text);
+	}
+	
+	public void switchToDefaultContent() {
+		this.driver.switchTo().defaultContent();
+	}
+	
+	public void switchToFrame(WebElement frame) {
+		this.driver.switchTo().frame(frame);
+	}
+	
+	public void hoverOverElementUsingJS(WebElement element) {
+		String js = "if(document.createEvent){var evObj = document.createEvent('MouseEvents');evObj.initEvent('mouseover',true, false); arguments[0].dispatchEvent(evObj);} else if(document.createEventObject) { arguments[0].fireEvent('onmouseover');}";
+		getJavaScriptExecutor().executeScript(js, element);
+	}
+	
+	public JavascriptExecutor getJavaScriptExecutor() {
+		if (javaScriptExecutor == null)
+			javaScriptExecutor = (JavascriptExecutor) driver;
+		return javaScriptExecutor;
+	}
+	
+	public void scrolltoElement(String locator) {
+		try {
+			WebElement element = this.driver.findElement(By.xpath(locator));
+
+			scrolltoElement(element);
+		} catch (Exception ex) {
+			
+		}
```

Avoid empty catch blocks, add proper codes to tackle the exceptions been appeared on runtime, if possible avoid using direct Exception class and add multiple level catch blocks handling different types of exceptions been specific to selenium WebElements or at least a proper log message should be added.